timemachine/fe/bar.py

```diff
@@ -142,27 +166,22 @@ def bootstrap_bar(w_F, w_R, n_bootstrap=1000, timeout=10) -> Tuple[float, NDArra
         if elapsed_time > timeout:
             break
 
-        w_F_sample = rng.choice(w_F, size=(n_F,), replace=True)
-        w_R_sample = rng.choice(w_R, size=(n_R,), replace=True)
+        u_kln_sample = rng.choice(u_kln, size=(n,), replace=True, axis=-1)
 
-        bar_result = pymbar.BAR(
-            w_F=w_F_sample,
-            w_R=w_R_sample,
-            DeltaF=full_bar_result,  # warm start
-            compute_uncertainty=False,
+        bar_result = df_from_u_kln(
+            u_kln_sample,
+            initial_f_k=mbar.f_k,  # warm start
             relative_tolerance=1e-6,  # reduce cost
```

Comment:
may need to tweak this (or add a TODO to tweak this in the future) -- I picked this in context of pymbar.BAR, not sure if it'll have similar behavior in context of pymbar.MBAR

Reply:
Could we use the built-in bootstrapping (even in 3.1.1) for MBAR? https://github.com/choderalab/pymbar/blob/3.1.1/pymbar/mbar.py#L316-L339

Maybe not for consideration for this PR, especially since the bootstrapping in pymbar3 was done in parallel with the bootstrapping for pymbar4 (https://github.com/choderalab/pymbar/pull/302 vs https://github.com/choderalab/pymbar/pull/448) so maybe they differ?

Reply:
Nevermind, ignore me. We are on 3.0.5 which doesn't have the bootstrapping.

Reply:
> Nevermind, ignore me. We are on 3.0.5 which doesn't have the bootstrapping.

Would like to return to this when we're able to upgrade to pymbar4 (this might actually be possible with the switch to MBAR, but haven't confirmed).

Reply:
> I picked this in context of pymbar.BAR, not sure if it'll have similar behavior in context of pymbar.MBAR

@maxentile can you elaborate on the behavior you'd like to verify here? Had you run some specific tests for the previous variant that we could repeat here?

Reply:
@mcwitt : Sorry, I don't have well-stated requirements, and I don't have specific tests available.

I think had eyeballed for a favorable trade-off between speeding things up and not changing the results too much,  relative to default solver tolerance https://github.com/proteneer/timemachine/pull/829#issuecomment-1247266763 . I think on instances of increasing difficulty of a gaussian work test system, but I did not retain good notes unfortunately.

Since the errors will be severely underestimated anyway (since sample autocorrelation is large and ignored), preserving the behavior of the default solver tolerance doesn't seem critical. Don't know if there's a big speed difference between 1e-6 and the default tolerance (1e-7).

In the pymbar references @badisa posted, I think the bootstrap solver tolerance in 3.1.1. may have been set to 1e-12(?), and in 4 it may have been left at 1e-7 but other solver settings were changed.

Reply:
@maxentile thanks! I'd actually misread the comment location and thought you were referring to the `initial_f_k` setting. Starting with the default tolerance of `1e-7` when we don't have reason to choose something different seems reasonable to me.

Reply:
Removed the override of the default tolerance setting in https://github.com/proteneer/timemachine/pull/1098/commits/5f93cf64ab8bc384b1d217902ae19a80a0219501

Reply:
>Starting with the default tolerance of 1e-7 when we don't have reason to choose something different seems reasonable to me.
>Removed the override of the default tolerance setting in https://github.com/proteneer/timemachine/commit/5f93cf64ab8bc384b1d217902ae19a80a0219501

Nice -- agreed!

(Can tweak later if we find a need to.)